src/editors/containers/TextEditor/index.jsx

```diff
@@ -40,6 +40,11 @@ const TextEditor = ({
     learningContextId,
   });
   const editorContent = newContent || initialContent;
+  let documentURL;
+  if (isLibrary) {
+    // TODO Get the library base URL in the format http://{STUDIO_HOST}/library_assets/{component_uuid}/
+    documentURL = 'http://studio.local.openedx.io:8001/library_assets/9b40a992-9d5f-4f22-9d8c-5ff18c3600dc/';
```

Comment:
How to get this URL correctly?

Reply:
In React, you can use [the `useXBlockAssets` hook](https://github.com/openedx/frontend-app-authoring/blob/4facf1cf5d91b259f8f324f562df4d0062548ae8/src/library-authoring/data/apiHooks.ts#L360-L367) to get the list of assets and their URLs. If not in React, you need to use the `getXBlockAssets` API method directly, but it won't be properly cached/de-duplicated/invalidated so it's much better to use the React Query `useXBlockAssets` hook if at all possible.

Reply:
Ok, I've implemented a working solution as the assets are copied over to every version, I'm just basically getting the whole list and reading the first element url and remove the specific asset path.

Reply:
The list may be empty though. Also, Learning Core doesn't necessarily guarantee that all the assets will have the same URL prefix (same path). That's just an implementation detail at the moment (or is it? @ormsbee). So it's better to get the URL for each asset from the list, and/or from the response when you upload the asset.

Reply:
I thought this would only show the draft version of the asset (aka: latest) which is what's returned by the API. And the problem is that this solution only works if all relative links to assets point to the same base path `{STUDIO_HOST}/library_assets/{component_uuid}/`. The preview works in the same way

Reply:
What @bradenmacdonald says is correct, but maybe this would be simpler if we added a new view that always points to the draft version of an asset and uses the usage key in the URL? It wouldn't be hard to add an endpoint like `library_assets/blocks/<str:usage_key_str>/<path:asset_path>'` that maps to a view that just does the UsageKey + Draft version -> component version UUID lookup and then delegates to `views.component_version_asset` to do all the work?

Reply:
Ok, lets do it. I will start working on that first thing Monday